src/treelearner/monotone_constraints.hpp

```diff
@@ -878,7 +878,7 @@ class AdvancedLeafConstraints : public IntermediateLeafConstraints {
     // the previous constraint that still applies needs to be recorded
     double previous_constraint;
     double current_constraint;
-    for (unsigned int i = 0; i < feature_constraint->thresholds.size();) {
+    for (size_t i = 0; i < feature_constraint->thresholds.size();) {
```

Comment:
why not `++i` inside? will this dangerous?
also, please replace `i += 1` to `++i` in the following codes.

Reply:
++i inside works, I had not cleaned this function very well.